### PR TITLE
introduce degraded condition

### PR DESCRIPTION
**Pull request by derekwaynecarr**

### CODE REVIEW

**Comment by derekwaynecarr** [comment]
related: https://github.com/openshift/cluster-version-operator/pull/161

/hold

**Comment by adambkaplan** [comment]
What about the consts defined in types_cluster_operator? https://github.com/openshift/api/blob/master/config/v1/types_cluster_operator.go#L121-L139

File: operator/v1/types.go [hunk 1]
```diff
@@ -135,6 +135,8 @@ var (
 	OperatorStatusTypeProgressing = "Progressing"
 	// Failing indicates that the operator (not the operand) is unable to fulfill the user intent
```
Comment (danehans) [inline operator/v1/types.go]: Does the Failing type comment need to be updated since it's the same as Degraded?

**Comment by eparis** [comment]
https://github.com/openshift/api/pull/287

**Pull request merged**